### PR TITLE
[guide] Add array-bracket-newline

### PR DESCRIPTION
**Pull request by JPeer264**
As discussed in #1338 I created a pull request. The naming, description and examples tend to the eslint rule `array-bracket-newline`, which hopefully comes soon (https://github.com/eslint/eslint/issues/6073):

```js
/*eslint array-bracket-newline: ["error", { "multiline": true }]*/
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -417,6 +417,45 @@ Other Style Guides
 
 **[⬆ back to top](#table-of-contents)**
 
+<a name="arrays--bracket-newline"></a><a name="4.5"></a>
```
Comment (ljharb) [inline README.md]: Numbered links are legacy, and should never be added or changed - let's remove the one on line 420 (but keep the visual number on line 421)

**Comment by JPeer264** [comment]
Done and squashed

**Comment by JPeer264** [comment]
@ljharb should I do a PR and remove all legacy numbered links throughout the README ?

**Comment by ljharb** [comment]
@JPeer264 no, the existing ones are kept permanently forever, intentionally.

**Pull request merged**